### ISSUE TITLE
Feature Request: stop music when time is up

### ISSUE THREAD

**Issue by NathanC** [moritzgloeckl/FallASleep#2]
First off, great app! The persistent notification is really well done.

It would be great to be able to stop music that is playing when the timer is finished. There is another fdroid app (not as developed) that slowly turns your volume down as it gets closer.

However, that would leave the audio service playing, which could drain battery. I'm not sure what permissions are needed to atop music or kill another app, but maybe one of those is viable? I wouldn't want to give it full notification access though.

tl;dr; I'm not sure of the implementation, but this would be a useful feature :)

**Comment by NathanC** [comment]
Since posting this, I found another sleep timer [here](https://github.com/SimonMarquis/SleepTimer), which quiets and then actually pauses the music at the end of the timer.  It doesn't require any notifications at all, outside of giving its own notification. I don't understand intents very well yet, but it looks like it's [asking for audio focus](https://github.com/SimonMarquis/SleepTimer/blob/main/app/src/main/java/fr/smarquis/sleeptimer/SleepAudioService.kt#L41), and giving up focus 2ms later, which presumably causes the music player to pause? Not sure of the exact mechanic.

Anyhow, I'm using that now so no pressure in implementing this feature, but if you decide to I hope the link helps :)
